File: main/contrib/Windows API Code Pack 1.1/source/Samples/AeroGlass/CS/WindowsFormsGlassDemo/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;

using Microsoft.WindowsAPICodePack.Shell;

namespace WindowsFormsGlassDemo
{
    public partial class Form1 : GlassForm
    {
        public Form1( )
        {
            InitializeComponent( );

            explorerBrowser1.Navigate( (ShellObject)KnownFolders.Desktop );

            AeroGlassCompositionChanged += new EventHandler<AeroGlassCompositionChangedEventArgs>( Form1_AeroGlassCompositionChanged );

            if( AeroGlassCompositionEnabled )
            {
                ExcludeControlFromAeroGlass( panel1 );
            }
            else
            {
                this.BackColor = Color.Teal;
            }

            // set the state of the Desktop Composition check box.
            compositionEnabled.Checked = AeroGlassCompositionEnabled;
        }

        void Form1_AeroGlassCompositionChanged( object sender, AeroGlassCompositionChangedEventArgs e )
        {
            // When the desktop composition mode changes the window exclusion must be changed appropriately.
            if( e.GlassAvailable )
            {
                compositionEnabled.Checked = true;
                ExcludeControlFromAeroGlass( panel1 );
                Invalidate( );
            }
            else
            {
                compositionEnabled.Checked = false;
                this.BackColor = Color.Teal;
            }
        }

        private void Form1_Resize( object sender, EventArgs e )
        {
            Rectangle panelRect = ClientRectangle;
            panelRect.Inflate( -30, -30 );
            panel1.Bounds = panelRect;
            ExcludeControlFromAeroGlass( panel1 );
        }

        private void compositionEnabled_CheckedChanged(object sender, EventArgs e)
        {
            // Toggles the desktop composition mode.
            AeroGlassCompositionEnabled = compositionEnabled.Checked;
        }
    }
}

File: main/contrib/Windows API Code Pack 1.1/source/Samples/AeroGlass/VB/WindowsFormsGlassDemo/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Imports Microsoft.WindowsAPICodePack.Shell

Namespace WindowsFormsGlassDemo
	Partial Public Class Form1
		Inherits GlassForm
		Public Sub New()
			InitializeComponent()

			explorerBrowser1.Navigate(CType(KnownFolders.Desktop, ShellObject))

			AddHandler AeroGlassCompositionChanged, AddressOf Form1_AeroGlassCompositionChanged

			If AeroGlassCompositionEnabled Then
				ExcludeControlFromAeroGlass(panel1)
			Else
				Me.BackColor = Color.Teal
			End If

			' set the state of the Desktop Composition check box.
			compositionEnabled.Checked = AeroGlassCompositionEnabled
		End Sub

        Private Sub Form1_AeroGlassCompositionChanged(ByVal sender As Object, ByVal e As AeroGlassCompositionChangedEventArgs)
            ' When the desktop composition mode changes the window exclusion must be changed appropriately.
            If e.GlassAvailable Then
                compositionEnabled.Checked = True
                ExcludeControlFromAeroGlass(panel1)
                Invalidate()
            Else
                compositionEnabled.Checked = False
                Me.BackColor = Color.Teal
            End If
        End Sub

		Private Sub Form1_Resize(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Resize
			Dim panelRect As Rectangle = ClientRectangle
			panelRect.Inflate(-30, -30)
			panel1.Bounds = panelRect
			ExcludeControlFromAeroGlass(panel1)
		End Sub

		Private Sub compositionEnabled_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles compositionEnabled.CheckedChanged
			' Toggles the desktop composition mode.
			AeroGlassCompositionEnabled = compositionEnabled.Checked
		End Sub
	End Class
End Namespace
